Userland/Libraries/LibWeb/HTML/WebViewHints.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2024, Andrew Kaster <akaster@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Optional.h>
+#include <AK/Types.h>
+#include <LibIPC/Forward.h>
+
+namespace Web::HTML {
+
+struct WebViewHints {
+    bool popup = false;
+    Optional<i32> width = {};
+    Optional<i32> height = {};
+    Optional<i32> screen_x = {};
+    Optional<i32> screen_y = {};
```

Comment:
If I stumbled across this, I'd wonder if these are CSS or device values. Maybe just make these:
```c++
Optional<DevicePixelSize> size;
Optional<DevicePixelPoint> screen_position;
```
(or the other way around) so that it's clear?

Reply:
They are not CSS values, as they shouldn't be scaled. I can use DevicePixels I suppose. these are part of really old and crusty window.open() options, so they aren't going to be very nice.

Reply:
Also we can independently specify width, height, x, and y. So not sure if using size/point is the right call?

Reply:
Got it - if this is existing confusion, then we can just leave it. Or make the 4 of them `DevicePixels`, your call.

Reply:
I meant old and crusty in the Web spec/web reality sense, not in the "our code" sense :P. I'll make these DevicePixels. I suspect this'll need refactored slightly anyway once we properly support applying these options from Window::open_impl anyway.